mhs/common/mhs_common/workflow/tests/test_synchronous.py

```diff
@@ -192,17 +196,67 @@ def setUp(self) -> None:
-                                                                                       correlation_id="qwe",
-                                                                                       interaction_details={},
-                                                                                       payload="nice message",
-                                                                                       work_description_object=None)
+        error, text, work_description_response = await self.wf.handle_outbound_message(
+            from_asid="202020",
+            message_id="123",
+            correlation_id="qwe",
+            interaction_details={'action': 'test-interaction'},
+            payload="nice message",
+            work_description_object=None)
 
         wdo.set_outbound_status.assert_called_with(work_description.MessageStatus.OUTBOUND_MESSAGE_RESPONSE_RECEIVED)
         self.assertEqual(error, 200)
         self.assertEqual(text, 'response body')
 
+    @mock.patch('mhs_common.workflow.synchronous.logger')
+    @mock.patch('mhs_common.state.work_description.create_new_work_description')
+    @async_test
+    async def test_success_audit_log_should_be_called_when_successful_response_is_returned_from_spine(self, wd_mock,
+                                                                                                      log_mock):
+        self._setup_success_workflow()
+        wdo = mock.MagicMock()
+        wdo.publish.return_value = test_utilities.awaitable(None)
+        wd_mock.return_value = wdo
+        wdo.set_outbound_status.return_value = test_utilities.awaitable(None)
+
+        await self.wf.handle_outbound_message(
+            from_asid="202020",
+            message_id="123",
+            correlation_id="qwe",
+            interaction_details={'action': 'test-interaction'},
+            payload="nice message",
+            work_description_object=None)
+
+        log_mock.audit.assert_called_with(
+            '0101',
+            'Outbound Synchronous workflow completed. Message sent to Spine and {Acknowledgment} received.',
+            {'Acknowledgment': True})
+
+    @mock.patch('mhs_common.workflow.synchronous.logger')
+    @mock.patch('mhs_common.state.work_description.create_new_work_description')
+    @async_test
+    async def test_initial_audit_log_should_be_called_handling_is_invoked(self, wd_mock, log_mock):
+        self._setup_success_workflow()
+        wdo = mock.MagicMock()
+        wdo.publish.return_value = test_utilities.awaitable(None)
+        wd_mock.return_value = wdo
+        wdo.set_outbound_status.return_value = test_utilities.awaitable(None)
+
+        future = asyncio.Future()
+        future.set_exception(httpclient.HTTPClientError(code=451))
+        self.transmission.make_request.return_value = future
+
+        await self.wf.handle_outbound_message(
+            from_asid="202020",
+            message_id="123",
+            correlation_id="qwe",
```

Comment:
Optional: this line seems to be very similar in these tests, and could possibly be refactored out.

Reply:
Possibly, though since it's a single method call anyway it would just end up being refactored into another method with the same number of parameters?

Reply:
Yes, but you can use keyword params with defaults which cuts down on duplication. See https://github.com/nhsconnect/integration-adaptors/blob/3fcfa40d2ae791f65ed8656e86b8b901c2beae30/mhs/outbound/outbound/request/synchronous/tests/test_handler.py#L33-L38 and other parts of that file where that method is used for an example of what I mean.

Bear in mind this comment is optional, I'll approve regardless.